snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -575,10 +576,7 @@ def prepare_build(self, force=False):
     def build(self, force=False):
         self.makedirs()
 
-        if not (
-            isinstance(self.plugin, plugins.v1.PluginV1)
-            and self.plugin.out_of_source_build
-        ):
+        if not self.plugin.out_of_source_build:
```

Comment:
for v2 we do not wipe anymore, to allow for faster iterative builds. Can that be taken into account? As @kyrofa mentions, might be better to just avoid the copy

Reply:
As @kyrofa mentioned in the other thread, isn't this covered by the existing `PluginHandler.build()` vs. `PluginHandler.update_build()` split?

Reply:
v2 falls under the _do_v2_build() logic, not this.  In fact, if a v2 were to execute this, the if would evaluate to True :D